### ISSUE TITLE
Trilinos packages have been deprecated (intended for removal in 2025)

### ISSUE THREAD

**Issue by mmuetzel** [ElmerCSC/elmerfem#569]
From upstream Trilinos version 16.0:
https://github.com/trilinos/Trilinos/blob/trilinos-release-16-0-branch/RELEASE_NOTES#L8-L30
> A large number of packages have been marked as deprecated in this release. We are planning to remove them in the fall of 2025. The list of packages is
>
>  - Amesos
>  - AztecOO
>  - Epetra
>  - EpetraExt
>  - Ifpack
>  - Intrepid
>  - Isorropia
>  - ML
>  - NewPackage
>  - Pliris
>  - PyTrilinos
>  - ShyLU_DDCore
>  - ThyraEpetraAdapters
>  - ThyraEpetraExtAdapters
>  - Triutils
>
> If you are a user of one of these packages, please transition to the newer software stack. If you need guidance with finding an adequate replacement or are missing a particular feature, please open an issue or a discussion on Github.

Afaict, at least the following of these are used by ElmerFEM (at least I had to enable them manually in Trilinos 16.0 so ElmerFEM didn't fail to build or link):
* AztecOO
* Epetra
* EpetraExt
* ML

**Comment by raback** [comment]
Trilinos has been the linear solver package with perhaps the least usage. In recent times effort has gone more to AMGx and rocALUTION that support also the GPU's. Hypre is also used more than Trilinos. 

Googling around of current Trilinos packages it seems that Epetra has been replace by Tpetra (Krylov methods), and ML by MueLu (algebraic multigrid). So whereas Trilinos seems to be useful we should work with the code to take use of these replacement. They also seem to offer more versatility in the backends.

**Comment by cgcgcg** [comment]
That's correct. Our newer solver stack (Tpetra, Belos, Ifpack2, MueLu, FROSch, etc) supports OpenMP, Cuda, HIP, SYCL, etc.
If you are indeed transitioning to the new stack, I would recommend to use the unified solver interface from the Stratimikos package. Here's an example of what that would look like:
https://github.com/trilinos/Trilinos/blob/master/packages/muelu/example/basic/Stratimikos.cpp

**Comment by raback** [comment]
The old references to the deprecated Trilinos solvers been removed for now. Updating to new stack may be done later. Then the unified solver interface sounds like the way to go.
